### PR TITLE
java: tiny bits of clean up to the host implementation

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
I think all the cases where `CopyDataToJava` is used it needs to be confirmed whether the object must be freed or not. In some cases it is freed, in others it is not.

**Review by atoulme — APPROVED** [review]
LGTM

File: bindings/java/c/host.c [hunk 1]
```diff
@@ -36,149 +36,122 @@ static jbyteArray CopyDataToJava(JNIEnv* jenv, const void* ptr, size_t size)
...
-    jaddress = CopyDataToJava(jenv, address, sizeof(struct evmc_address));
+    jbyteArray jaddress = CopyDataToJava(jenv, address, sizeof(struct evmc_address));
 
     // call java method
-    jint jresult = (*jenv)->CallStaticIntMethod(jenv, host_class, method, jcontext_index, jaddress);
-    result = !!jresult;
-
-    return result;
+    jint jresult = (*jenv)->CallStaticIntMethod(jenv, host_class, method, context->index, jaddress);
+    return jresult == 1;
```
Comment (chfast) [inline bindings/java/c/host.c:59]:
```suggestion
    return jresult != 0;
```
More classic "convert int to bool".

Comment (axic) [reply bindings/java/c/host.c]: Fixed.

**Review by chfast — APPROVED** [review]

**Pull request merged**